src/AppBundle/Helper/AutomatedEditsHelper.php

```diff
@@ -65,13 +78,42 @@ public function isAutomated(string $summary, Project $project): bool
         return (bool)$this->getTool($summary, $project);
     }
 
+    /**
+     * Fetch the config from https://meta.wikimedia.org/wiki/MediaWiki:XTools-AutoEdits.json
+     * @param bool $useSandbox
+     * @return array
+     */
+    public function getConfig(bool $useSandbox = false): array
+    {
+        $cacheKey = 'autoedits_config';
+        if (!$useSandbox && $this->cache->hasItem($cacheKey)) {
+            return $this->cache->getItem($cacheKey)->get();
+        }
+
+        $title = 'MediaWiki:XTools-AutoEdits.json' . ($useSandbox ? '/sandbox' : '');
+        $ret = json_decode(file_get_contents(
+            $this->metaProject->getUrl() . "w/index.php?action=raw&ctype=application/json&title=$title"
+        ), true);
+
+        if (!$useSandbox) {
+            $cacheItem = $this->cache
```

Comment:
Is there any way to invalidate this when the page is edited?

Reply:
Not without querying the db or API (probably would need to be the letter) on every request. Not worth it in my opinion, as we're only talking up to a 20 min wait.

Reply:
(noting here what I said on IRC) I meant a way to purge the cache from within the xtools interface

Reply:
https://phabricator.wikimedia.org/T209334